app/models/events/user_agent_info.rb

```diff
@@ -0,0 +1,36 @@
+class Events::UserAgentInfo < OIDC::BaseModel
+  attribute :installer, :string
+  attribute :device, :string
+  attribute :os, :string
+  attribute :user_agent, :string
+  attribute :implementation, :string
+  attribute :system, :string
+
+  def to_s # rubocop:disable Metrics
+    if installer == "Browser"
+      if device != "Other" && os != "Other" && user_agent != "Other"
+        "#{user_agent} (#{os} on #{device})"
+      elsif device != "Other" && user_agent != "Other"
+        "#{user_agent} (#{device})"
+      elsif os != "Other" && user_agent != "Other"
+        "#{user_agent} (#{os})"
+      elsif user_agent != "Other"
+        user_agent
+      else
+        I18n.t("#{model_name.i18n_key}.unknown_browser")
+      end
+    elsif installer.present?
+      if implementation.present? && system.present?
+        "#{installer} (#{implementation} on #{system})"
+      elsif implementation.present?
+        "#{installer} (#{implementation})"
+      elsif system.present?
+        "#{installer} (#{system})"
+      else
+        installer
+      end
+    else
+      I18n.t("#{model_name.i18n_key}.unknown")
```

Comment:
I'm not sure it makes sense to I18n because the logs will be permanent and we don't want the logging to depend on... what even is the logging going to depend on? Will it depend on the user agent's locale settings in this particular request?

It's either going to be all English or it's going to switch to a new locale when we change the default locale or worse. I certainly don't want it to log "unknown browser" in whatever language the user has set at the time.